packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -5453,6 +5453,34 @@ class Flow extends MultiChildRenderObjectWidget {
 /// ```
 /// {@end-tool}
 ///
+/// ## Selections
+///
+/// To make this [RichText] Selectable, the [RichText] needs to be in the
+/// subtree of a [SelectionArea] or [SelectableRegion] and assigned a
+/// [SelectionRegistrar] to the [RichText.selectionRegistrar]. One can use
+/// [SelectionRegistrarScope.maybeOf] to get the [SelectionRegistrar] from a
+/// context. This enable users to select the texts in [RichText]s with mice or
+/// touches.
+///
+/// The [selectionColor] also needs to be set if the selection is enabled to
+/// draw the selection highlights.
+///
+/// {@tool snippet}
+///
+/// This sample demonstrates how to assign selection registrar for RichTexts
+/// in the SelectionArea subtree.
+///
+/// ![](https://flutter.github.io/assets-for-api-docs/assets/widgets/rich_text.png)
+///
+/// ```dart
+/// RichText(
+///   text: const TextSpan(text: 'Hello'),
+///   selectionRegistrar: SelectionRegistrarScope.maybeOf(context),
```

Comment:
Why is this needed for RichText but not for Text?

Reply:
Text is smarter than RichText, it will grab the selection registrar/ selection style/ mouse region.  RichText does not automatically imply things.

I can make RichText smarter, but then two thing would happen

1. It can no longer be a RenderObjectWidget because of mouseRegion.
2. customers wouldn't be able to customize these selection behaviors.

Reply:
Got it, I'm on board with it as-is.